pkg/apis/kubeone/validation/validation_test.go

```diff
@@ -2090,37 +2089,18 @@ func TestValidateAddons(t *testing.T) {
 		{
 			name: "valid addons config (enabled)",
 			addons: &kubeoneapi.Addons{
-				Enable: true,
-				Path:   "./addons",
+				Path: "./addons",
 			},
 			expectedError: false,
 		},
 		{
-			name: "addons enabled, no path set and no embedded addons specified",
-			addons: &kubeoneapi.Addons{
-				Enable: true,
-				Path:   "",
-			},
-			expectedError: true,
-		},
-		{
-			name: "embedded addon enabled, no path set",
-			addons: &kubeoneapi.Addons{
-				Enable: true,
-				Path:   "",
-				Addons: []kubeoneapi.Addon{
-					{
-						Name: resources.AddonMachineController,
-					},
-				},
-			},
+			name:          "no path set and no embedded addons specified",
+			addons:        &kubeoneapi.Addons{},
 			expectedError: false,
 		},
 		{
-			name: "valid addons config (disabled)",
-			addons: &kubeoneapi.Addons{
-				Enable: false,
-			},
+			name:          "valid addons config (disabled)",
```

Comment:
This is the same case as `no path set and no embedded addons specified`, we can remove this one.